lib/spack/spack/cmd/develop.py

```diff
@@ -80,8 +91,14 @@ def _retrieve_develop_source(spec, abspath):
 
 
 def develop(parser, args):
+    # TODO: when https://github.com/spack/spack/pull/35307 is merged,
+    # an active env is not required if a scope is specified
+    env = spack.cmd.require_active_env(cmd_name="develop")
+    if args.always_rebuild:
+        with env.write_transaction():
+            spack.config.add("config:dev_specs_always_rebuild:true", env.scope_name)
```

Comment:
This belongs in the develop section (and I think it needs to be set per-package, see above).